### PR TITLE
DAGCircuit._copy_circuit_metadata is widely used and it should be public

### PR DESCRIPTION
**Pull request by 1ucian0**
It seems like several passes are using `DAGCircuit._copy_circuit_metadata`:
```
$ grep -rI _copy_circuit_metadata qiskit/transpiler/  | wc
       7      28     746
```
and [another one coming](https://github.com/Qiskit/qiskit-terra/pull/7725#discussion_r832278150).

Since the method is public _de facto_, I think it is time to make it official.

### CODE REVIEW

File: qiskit/dagcircuit/dagcircuit.py [hunk 1]
```diff
@@ -496,7 +496,7 @@ def _add_op_node(self, op, qargs, cargs):
         self._increment_op(op)
         return node_index
 
-    def _copy_circuit_metadata(self):
+    def copy_circuit_metadata(self):
         """Return a copy of source_dag with metadata but empty."""
```
Comment (mtreinish) [inline qiskit/dagcircuit/dagcircuit.py:500]: Since this is public now, do you think we should expand this docstring a bit (since it will be published)? Maybe document which attributes are copied or something.

Comment (1ucian0) [reply qiskit/dagcircuit/dagcircuit.py:500]: actually, now we have a metadata attribute. Maybe we can call the method `copy_empty`?

Comment (mtreinish) [reply qiskit/dagcircuit/dagcircuit.py]: I personally find `empty_like` a bit confusing, since there is no verb in the method name I wouldn't know what it's doing (my first guess would be it's a boolean check to see if the number of elements was near zero). What about `empty_like_copy` or `copy_empty_like`?

Comment (1ucian0) [reply qiskit/dagcircuit/dagcircuit.py]: Ill go for `copy_empty_like`

File: test/python/dagcircuit/test_dagcircuit.py [hunk 2]
```diff
@@ -254,6 +254,8 @@ class TestDagWireRemoval(QiskitTestCase):
     def setUp(self):
         super().setUp()
         self.dag = DAGCircuit()
+        self.name = "Name"
+        self.name = "Metadata"
```
Comment (Cryoris) [inline test/python/dagcircuit/test_dagcircuit.py:258]: Are these changes required?

File: qiskit/dagcircuit/dagcircuit.py [hunk 3]
```diff
@@ -491,8 +491,18 @@ def _add_op_node(self, op, qargs, cargs):
...
+        """Return a copy of self with the same structure but empty.
+
+        That structure includes:
+            * name and other metadata
+            * global phase
+            * duration
+            * all the qubits and clbits, including the registers.
+
+        Returns:
+            DAGCircuit: An empty copy of self but empty.
```
Comment (Cryoris) [inline qiskit/dagcircuit/dagcircuit.py:504]:
```suggestion
            DAGCircuit: An empty copy of self.
```

File: qiskit/dagcircuit/dagcircuit.py [hunk 4]
```diff
@@ -491,8 +491,18 @@ def _add_op_node(self, op, qargs, cargs):
         self._increment_op(op)
         return node_index
 
-    def _copy_circuit_metadata(self):
-        """Return a copy of source_dag with metadata but empty."""
+    def copy_empty_like(self):
```
Comment (mtreinish) [inline qiskit/dagcircuit/dagcircuit.py:494]: If we're changing the name here I wonder if it makes sense to add an alias at the old name and raise a deprecation warning. While it wasn't public as you say in the commit message summary line it was widely used. So having something like:

```python
def _copy_circuit_metadata(self):
    warnings.warn("deprecated", DeprecationWarning)
    return self.copy_empty_like()
```

isn't too terrible to keep around for a release or two. Since it was private I think forgoing the requirement that the alternative/replacement is available in a release prior to the deprecation release (`2a` in: https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-policy ) is probably ok.

File: test/python/dagcircuit/test_dagcircuit.py (hunk 2 above)
Comment (1ucian0) [reply test/python/dagcircuit/test_dagcircuit.py:258]: ups! bad copypasta!
 
```suggestion
        self.name = "Name"
        self.metadata = "Metadata"
```

File: qiskit/dagcircuit/dagcircuit.py (hunk 4 above)
Comment (1ucian0) [reply qiskit/dagcircuit/dagcircuit.py:494]: I agree. Added in aaaffaa19c1aa656a40fa720cc2c4e6ad9bd5d69

**Review by Cryoris — APPROVED** [review]
LGTM!
